docs/topics/multiplatform/multiplatform-hierarchy.md

```diff
@@ -116,114 +131,106 @@ to your gradle.properties
-    // The rest of the other targets, if needed.
-
-    sourceSets {
-        commonMain {
-          
-        }
-
-        jvmAndNativeMain {
-            dependsOn(commonMain)
-        }
-
-        nativeMain {
-            dependsOn(jvmAndNativeMain)
-        }
-
-        jvmMain {
-            dependsOn(jvmAndNativeMain)
+    <tabs group="build-script">
+    <tab title="Kotlin" group-key="kotlin">
+    
+    ```kotlin
+    kotlin {
+        jvm()
+        macosArm64()
+        iosArm64()
+        iosSimulatorArm64()
+    
+        // Apply the default hierarchy again. It'll create, for example, the iosMain source set:
+        applyDefaultHierarchyTemplate()
+    
+        sourceSets {
+            // Create an additional jvmAndMacos source set:
+            val jvmAndMacos by creating {
+                dependsOn(commonMain.get())
+            }
+    
+            macosArm64Main.get().dependsOn(jvmAndMacos)
+            jvmMain.get().dependsOn(jvmAndMacos)
         }
     }
-}
-```
-
-</tab>
-</tabs>
+    ```
+    
+    </tab>
+    <tab title="Groovy" group-key="groovy">
+    
+    ```groovy
+    kotlin {
+        jvm()
+        macosArm64()
+        iosArm64()
+        iosSimulatorArm64()
+    
+        // Apply the default hierarchy again. It'll create, for example, the iosMain source set:
+        applyDefaultHierarchyTemplate()
+    
+        sourceSets {
+            // Create an additional jvmAndMacos source set:
+            jvmAndMacos {
+                dependsOn(commonMain.get())
+            }
+            jvmAndMacos  {
+                dependsOn(macosArm64Main.get())
+            }
+            jvmAndMacos  {
+                dependsOn(jvmMain.get())
+            }
+        } 
+    }
+    ```
+    
+    </tab>
+    </tabs>
 
-It can be cumbersome to remove `dependsOn` relations that are automatically created by the `targetHierarchy.default()` call.
-In that case, use an entirely [manual configuration](#manual-configuration) instead of calling the default hierarchy.
+#### Modifying source sets
 
-#### Modifying the source sets created by the Default Hierarchy Template
+**Case**. You already have the source sets with the exact same names as those generated by the template but shared among
+different sets of targets in your project. For example, a `nativeMain` source set shared among not all the native 
+targets but only the desktop ones: `linuxX64`, `mingwX64`, and `macosX64`.
 
-**Case.** You want to have a source set exactly with the same name as the one created by the Default Target Hierarchy,
-but shared for a different set of targets. For example, you want to have a `nativeMain` shared not for all native 
-targets, but only for the desktop ones: `linuxX64`, `mingwX64`, `macosX64`.
+**Solution**. There's currently no way to modify the default `dependsOn` relations among the template's source sets.
+It's a technical limitation of the template. It's also important that the implementation and the meaning of source sets,
```

Comment:
> It's a technical limitation of the template

It is true to some extent, but what I wanted to convey here is actually the opposite: it's not *only* a technical limitation, we actually *do not like* the idea of allowing this from the design perspective. Let's remove this sentence maybe?